include/knowrob/BaseError.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2024, Sascha Jongebloed
+ * All rights reserved.
+ *
+ * This file is part of KnowRob, please consult
+ * https://github.com/knowrob/knowrob for license details.
+ */
+
+#ifndef KNOWROB_ERROR_H_
+#define KNOWROB_ERROR_H_
+
+#include <string>
+#include <exception>
+#include <fmt/core.h>
+
+namespace knowrob {
+
+	/**
+	 * A knowrob runtime error with optional stacktrace.
+	 */
+	class BaseError : public std::runtime_error {
+	private:
+		std::string stacktrace; // Field to store the stacktrace
+
+	public:
+		/**
+		 * Constructor for error without stacktrace.
+		 * @tparam Args fmt-printable arguments.
+		 * @param fmt A fmt string pattern.
+		 * @param args list of arguments used to instantiate the pattern.
+		 */
+		template<typename ... Args>
+		explicit BaseError(const char *fmt, Args&& ... args)
+				: std::runtime_error(fmt::format(fmt, std::forward<Args>(args)...))
+		{}
+
+		/**
+		 * Constructor for error with stacktrace.
+		 * @tparam Args fmt-printable arguments.
+		 * @param fmt A fmt string pattern.
+		 * @param stacktrace String representing the captured stack trace.
+		 * @param args list of arguments used to instantiate the pattern.
+		 */
+		template<typename ... Args>
+		explicit BaseError(const char *fmt, const std::string& stacktrace, Args&& ... args)
```

Comment:
it is a bit odd that the fmt format string is not directly next to the args. Because usually you write something like `"foo {}", val` and would assume val gets inserted into the format.

It's not possible to have it after `args`?